### ISSUE TITLE
Random greek alphabet letter

### ISSUE THREAD

**Issue by Shinigami92** [faker-js/faker#558]
### Clear and concise description of the problem

As a user I want to generate a random greek letter so that I can prepend this to a name of an object.

I would like to have the option to get a short or a long version of a letter (`α` or `'alpha'`).
Maybe `Α` and `άλφα` could also somehow get returned.

### Suggested solution

```ts
const cameraName = 'Camera ' + faker.<?>.greekLetter({ format: 'long' });

const cameraName2 = faker.helpers.mustache('Camera {{greekLetter({format:"long"})}}');
```

I'm open to other ideas

### Additional context

https://en.wikipedia.org/wiki/Greek_alphabet#Letters

**Comment by ST-DDT** [comment]
AFAICT you use the greek characters to name or enumerate things. Why dont you use numbers or letters instead?

If we add support for greek characters, what about russian letters or chinese word letters?

**Comment by ST-DDT** [comment]
Potentially changed requirements:

`faker.xyz.letter({ case: lower | upper | name })`

lower =>'a'
upper => 'A'
name => 'Aey'(?)

TBD: The name in which locale? Alpha vs άλφα.

**Comment by matthewmayer** [comment]
I don't think this should require an el locale. Greek letters are often used when a neutral name is needed (eg think of covid variants)

**Comment by ST-DDT** [comment]
How about `faker.string.letter({ alphabet: greek })`?

**Comment by ST-DDT** [comment]
Or adding an alphabet parameter to string.alpha?

**Comment by ST-DDT** [comment]
As a workaround, you can use
`faker.string.fromCharacters('α...ω')` or `faker.string.fromCharacters('Α...Ω')`

We could simplify this some more, by either providing constants like `GREEK_LETTER_LOWERCASE` or adding another signature to `fromCharacter(charRanges: Array<[start: string, end: string]>, ...)`

**Comment by Shinigami92** [comment]
It was not my intend to build up a huge discussion about how faker works, and how the structure for a simple new feature, that could ship improvement for customer, can be over-complexified.
It's a bit sad to see that it is not at all possible in an easy way to first ship an improvement and enhance from there, but just get massive road-blocks set into the way :disappointed:  
So instead I might just go on with just `faker.helpers.arrayElement('α...ω')`, like I already do with some other things.
I just hoped I could add abilities to faker to set a common base of getting date.

**Comment by xDivisionByZerox** [comment]
> It's a bit sad to see that it is not at all possible in an easy way to first ship an improvement and enhance from there, but just get massive road-blocks set into the way 😞

Sorry, but this is exactly the attitude that led to an entire major version being used for restructuring the API for our users. Stability is a major keystone in developing well maintained software that is used by external users. The way you suggest to work, plan, and ship software simply goes against these basic principles. You make it seem as if discussion would be a bad thing. On top of that, the discussion isn't even that big right now.

**Comment by Shinigami92** [comment]
> > It's a bit sad to see that it is not at all possible in an easy way to first ship an improvement and enhance from there, but just get massive road-blocks set into the way 😞
> 
> Sorry, but this is exactly the attitude that led to an entire major version being used for restructuring the API for our users. Stability is a major keystone in developing well maintained software that is used by external users. The way you suggest to work, plan, and ship software simply goes against these basic principles. You make it seem as if discussion would be a bad thing. On top of that, the discussion isn't even that big right now.

It's just like always when I try something add to Faker and try to make progress, everyone jumps in and brings hard-breaking counter ideas out of the sudden.
That does not make fun anymore to work for Faker if there is always such a discussion.

**Comment by matthewmayer** [comment]
> It was not my intend to build up a huge discussion about how faker works, and how the structure for a simple new feature, that could ship improvement for customer, can be over-complexified. It's a bit sad to see that it is not at all possible in an easy way to first ship an improvement and enhance from there, but just get massive road-blocks set into the way 😞 So instead I might just go on with just `faker.helpers.arrayElement('α...ω')`, like I already do with some other things. I just hoped I could add abilities to faker to set a common base of getting date.

(more huge discussion, sorry 🤣)

I totally acknowledge @Shinigami92's feelings on this topic, I think it's very understandable to feel frustrated at the pace of change in a project like this.

But, as others have mentioned, I think that making it "hard" to make quick additions to the API is done with good reasons. 
If you look at where Faker is now, it seems in a much healthier state than a few years ago, and that's primarily because of having a diverse group of maintainers who don't always agree on the right way to do things. We all care about different things, whether it be how to name an API, whether the API should even belong in Faker, Typescript vs Javascript, documentation, examples, comments, localization. That diversity is generally a good thing. Occasionally it will lead to long rambling discussions, verging on bikeshedding. 

I recently was looking at Faker-Ruby and while its also an excellent project, which has way more data than we have, its been added to very haphazardly over the years, so the structure of the API and the documentation is far messier than Faker JS, to the extent they actively ask people not to contribute new data on the CONTRIBUTING page.

Because of semver, it's important to make an effort to get the API "right" first time. The end result is a much cleaner and more logical API, which is easy for developers to understand and use.

**Comment by Shinigami92** [comment]
I will live with the suggested "workaround" for now

- https://github.com/faker-js/faker/issues/558#issuecomment-2067999050 (`fromCharacters`)
- https://github.com/faker-js/faker/issues/558#issuecomment-2068087454 (`arrayElement` the variant I already used in my code)

Also I'm tempted to ship never self-requested features again to Faker: https://github.com/faker-js/faker/pull/2836#issuecomment-2068632206

**Comment by matthewmayer** [comment]
if anyone is stumbling on this issue, this will work to give you a single character:
```
faker.string.fromCharacters("ΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ")
faker.string.fromCharacters("αβγδεζηθικλμνξοπρστυφχψω")
```
